src/web/client/common/localStore.ts

```diff
@@ -23,56 +25,62 @@ export interface IPowerPlatformExtensionContext {
     entity: string;
     rootDirectory: vscode.Uri;
     saveDataMap: Map<string, SaveEntityDetails>,
-    defaultFileUri: vscode.Uri // This will default to home page or current page in multifile scenario
+    defaultFileUri: vscode.Uri, // This will default to home page or current page in multifile scenario
+    contextSet: boolean,
+    currentSchema: string
 }
 
 class PowerPlatformExtensionContextManager {
-    private entitiesSchemaMap = getEntitiesSchemaMap();
-    private dataSourcePropertiesMap = getDataSourcePropertiesMap();
-    private entitiesFolderNameMap = getEntitiesFolderNameMap(this.entitiesSchemaMap);
 
     private PowerPlatformExtensionContext: IPowerPlatformExtensionContext = {
-        dataSourcePropertiesMap: this.dataSourcePropertiesMap,
-        entitiesSchemaMap: this.entitiesSchemaMap,
+        dataSourcePropertiesMap: new Map<string, string>(),
+        entitiesSchemaMap: new Map<string, Map<string, string>>(),
         languageIdCodeMap: new Map<string, string>(),
         websiteLanguageIdToPortalLanguageMap: new Map<string, string>(),
         websiteIdToLanguage: new Map<string, string>(),
         queryParamsMap: new Map<string, string>(),
-        entitiesFolderNameMap: this.entitiesFolderNameMap,
+        entitiesFolderNameMap: new Map<string, string>(),
         entity: '',
         entityId: '',
         dataverseAccessToken: '',
         rootDirectory: vscode.Uri.parse(''),
         saveDataMap: new Map<string, SaveEntityDetails>(),
-        defaultFileUri: vscode.Uri.parse(``)
+        defaultFileUri: vscode.Uri.parse(``),
+        contextSet: false,
+        currentSchema: ""
     };
 
     public getPowerPlatformExtensionContext() {
         return this.PowerPlatformExtensionContext;
     }
 
-    public async setPowerPlatformExtensionContext(pseudoEntityName: string, entityId: string, queryParamsMap: Map<string, string>) {
-        this.PowerPlatformExtensionContext = {
-            ...this.PowerPlatformExtensionContext,
-            entity: pathParamToSchema.get(pseudoEntityName) as string,
-            entityId: entityId,
-            queryParamsMap: queryParamsMap,
-            rootDirectory: vscode.Uri.parse(`${PORTALS_URI_SCHEME}:/${queryParamsMap.get(WEBSITE_NAME) as string}/`, true)
-        };
-
-        return this.PowerPlatformExtensionContext;
+    public async setPowerPlatformExtensionContext(entityName: string, entityId: string, queryParamsMap: Map<string, string>) {
```

Comment:
Rename to setPowerPagesContext? This context is specific to portals and not related with the overall extension

Reply:
Not really. This is a extension wide object, and should be used by other apps onboarding to our extension for that purpose.